monkey/monkey_island/cc/services/config.py

```diff
@@ -235,9 +239,34 @@ def init_default_config():
     def get_default_config(should_encrypt=False):
         ConfigService.init_default_config()
         config = copy.deepcopy(ConfigService.default_config)
+
         if should_encrypt:
             ConfigService.encrypt_config(config)
+
         logger.info("Default config was called")
+
+        return config
+
+    @staticmethod
+    def update_config_on_mode_set(mode: str) -> None:
+        config = ConfigService.get_config()
+        ConfigService.update_config_per_mode(mode, config, True)
+
+    @staticmethod
+    def update_config_per_mode(mode: str, config: Dict, should_encrypt: bool) -> None:
+        config = ConfigService._set_default_config_values_per_mode(mode, config)
+        ConfigService.update_config(config_json=config, should_encrypt=True)
+
+    @staticmethod
+    def _set_default_config_values_per_mode(mode: str, config: Dict) -> Dict:
+        config_filter = FILTER_PER_MODE[mode]
+        config = ConfigService._apply_config_filter(config, config_filter)
+        return config
+
+    @staticmethod
+    def _apply_config_filter(config: Dict, config_filter: Dict):
```

Comment:
Filter would filter something out, this looks like it's setting the value. I used the term `template` in BB code, but maybe we can come up with something better.

Reply:
`_apply_config_manipulator`?